runelite-client/src/test/java/net/runelite/client/plugins/timers/TimersPluginTest.java

```diff
@@ -233,4 +242,65 @@ public void testEndurance()
 		int mins = (int) infoBox.getDuration().toMinutes();
 		assertTrue(mins == 1 || mins == 2);
 	}
+
+	@Test
+	public void testTzhaarTimer()
+	{
+		when(timersConfig.showTzhaarTimers()).thenReturn(true);
+		when(client.getMapRegions()).thenReturn(new int[]{FIGHT_CAVES_REGION_ID});
+		ChatMessage chatMessage = new ChatMessage(null, ChatMessageType.GAMEMESSAGE, "", TZHAAR_WAVE1_MESSAGE, "", 0);
+		timersPlugin.onChatMessage(chatMessage);
+
+		// test timer creation
+		ArgumentCaptor<InfoBox> captor = ArgumentCaptor.forClass(InfoBox.class);
+		verify(infoBoxManager).addInfoBox(captor.capture());
+		try
+		{
+			Thread.sleep(1000);
+		}
+		catch (Exception e)
+		{
+			log.debug("Thread interrupted", e);
+		}
+
+		// test timer pause
+		chatMessage = new ChatMessage(null, ChatMessageType.GAMEMESSAGE, "", TZHAAR_PAUSED_MESSAGE, "", 0);
+		timersPlugin.onChatMessage(chatMessage);
+		verify(infoBoxManager).removeInfoBox(captor.capture());
+		captor = ArgumentCaptor.forClass(InfoBox.class);
+		verify(infoBoxManager, times(2)).addInfoBox(captor.capture());
+		ElapsedTimer timer = (ElapsedTimer) captor.getValue();
+		assertEquals("00:01", timer.getText());
```

Comment:
The elapsed time testing should be part of a dedicated ElapsedTimerTest suite. I'm not convinced it's a good idea to test this anyway since this explicitly requires 3 seconds of CPU time to wait on sleeping threads. In any event this test case should just be ensuring appropriate responses to expected chat messages.

Reply:
How does one "ensure appropriate responses" if the test doesn't let the timer run? Like, if you don't create an actual timer how do you test to see if it pauses correctly or not? I made the test per adam's recommendations:
> I would test 1) create on first wave 2) pause it and see if it pauses 3) unpause it and see if it unpauses 4) die and see if it clears at the very least
> eg. send it a start wave message, capture the value passed to addInfoBox() and 1) check it is a elapsed timer 2) check the startTime/lastTime etc of that timer is something you expect

Reply:
If anything, the time comparisons should be in an ElapsedTimerTest suite. The plugin should not test that the timer works--only that its state is as expected at various times. So for the examples given:

1) Create on first wave: verify the infobox was added and that it is an ElapsedTimer
2) Pause it and see if it pauses: verify the added ElapsedTimer has a non-null lastTime
3) Unpause it and see if it unpauses: save the result from the paused timer, verify that the start time is not equal after being unpaused
4) Die and see if it clears: verify that the infobox was removed (and no more were added)